### ISSUE TITLE
Add methods consuming Collection for Subscribe/UnsubscribeBuilder

### ISSUE THREAD

**Issue by SgtSilvio** [hivemq/hivemq-mqtt-client#170]

**Comment by MicWalter** [comment]
**Motivation:**
It should be easy to add a dynamic amount of topics to a Subscribe/Unsubscribe packet. 

**Problem:**
At the moment for each Mqtt5Subscription/Mqtt5Unsubscribe you need to call addSubscription()/addTopicFilter(), that means you can't easily create a Subscribe/Unsubscribe packet that has a random amount of subscriptions/unsubscriptions.

Example:
Subscribe
 ```
final Mqtt5Subscribe subscribeMessage = Mqtt5Subscribe.builder()
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .addSubscription(Mqtt5Subscription.builder().topicFilter(topic).qos(MqttQoS.AT_LEAST_ONCE).build())
                .build();
```

Unsubscribe:
```
        subscriber.unsubscribe(Mqtt5Unsubscribe.build()
                .addTopicFilter("unsub1")
                .addTopicFilter("unsub2")
                .addTopicFilter("unsub3")
                .addTopicFilter("unsub4")
                .addTopicFilter("unsub5")
                .addTopicFilter("unsub6")
                .build())
                .doOnSuccess(success -> {
                    System.out.println("unsuback success; reason code: " + success.getReasonCodes().toString());
                })
                .doOnError(error -> System.out.println("error"))
                .subscribe();
```

**Proposed changes:**
You can give the methods addSubscription()/addTopicFilter() a List as parameter.

Example:
Subscribe
```
        ImmutableList<Mqtt5Subscribe> topics = createTopicsForRole(clientID);

        final Mqtt5Subscribe subscribeMessage = Mqtt5Subscribe.builder()
                .addSubscription(topics)
                .build();
```

Unsubscribe
```
        ImmutableList<String> unsubTopics = getTopicsToUnsubscribeForNewGeoPosition();

        subscriber.unsubscribe(Mqtt5Unsubscribe.build()
                .addTopicFilter(unsubTopics)
                .build())
                .doOnSuccess(success -> {
                    System.out.println("Removed obsolete geo position topics; reason code: " + success.getReasonCodes().toString());
                })
                .doOnError(error -> System.out.println("error"))
                .subscribe();

```

**Comment by SgtSilvio** [comment]
Method names:
- SubscribeBuilder: `addSupscriptions`
- UnsubscribeBuilder: `addTopicFilters`

Also add an `addAll` method to `Mqtt5UserPropertiesBuilder`

**Comment by SgtSilvio** [comment]
Resolved by #290
